tests/rule_based_profiler/parameter_builder/test_parameter_builder.py

```diff
@@ -0,0 +1,485 @@
+    )
+
+    my_parameter_builder.resolve_evaluation_dependencies(
+        domain=domain,
+        variables=None,
+        parameters=parameters,
+        fully_qualified_parameter_names=None,
+        runtime_configuration=None,
+    )
+
+    all_fully_qualified_parameter_names: List[
+        str
+    ] = get_fully_qualified_parameter_names(
+        domain=domain,
+        variables=None,
+        parameters=parameters,
+    )
+
+    """
+    This assertion shows that shared memory is empty: no "ParameterBuilder" dependencies given to compute values of.
+    """
+    assert not all_fully_qualified_parameter_names
+
+
+def test_resolve_evaluation_dependencies_two_parameter_builder_dependencies_specified(
+    empty_rule_state: Dict[str, Union[Domain, Dict[str, ParameterContainer]]],
+):
+    domain = empty_rule_state["domain"]
+    parameters = empty_rule_state["parameters"]
+
+    my_evaluation_dependency_0_parameter_builder: ParameterBuilder = (
+        DummyParameterBuilder(
+            name="my_evaluation_dependency_parameter_name_0",
+            evaluation_parameter_builder_configs=None,
+        )
+    )
+    my_evaluation_dependency_1_parameter_builder: ParameterBuilder = (
+        DummyParameterBuilder(
+            name="my_evaluation_dependency_parameter_name_1",
+            evaluation_parameter_builder_configs=None,
+        )
+    )
+
+    evaluation_parameter_builder_configs: Optional[List[ParameterBuilderConfig]] = [
+        ParameterBuilderConfig(
+            **my_evaluation_dependency_0_parameter_builder.to_json_dict()
+        ),
+        ParameterBuilderConfig(
+            **my_evaluation_dependency_1_parameter_builder.to_json_dict(),
+        ),
+    ]
+    my_dependent_parameter_builder: ParameterBuilder = DummyParameterBuilder(
+        name="my_dependent_parameter_builder",
+        evaluation_parameter_builder_configs=evaluation_parameter_builder_configs,
+    )
+
+    my_dependent_parameter_builder.resolve_evaluation_dependencies(
+        domain=domain,
+        variables=None,
+        parameters=parameters,
+        fully_qualified_parameter_names=None,
+        runtime_configuration=None,
+    )
+
+    all_fully_qualified_parameter_names: List[
+        str
+    ] = get_fully_qualified_parameter_names(
+        domain=domain,
+        variables=None,
+        parameters=parameters,
+    )
+
+    """
+    This assertion shows that because "ParameterBuilder" dependencies were specified as argument to instantiation of
+    dependent "ParameterBuilder", both "ParameterBuilder" dependencies computed their values (this was accomplished by
+    "my_dependent_parameter_builder.resolve_evaluation_dependencies()", which ran "build_parameters()" method of each
+    dependency "ParameterBuilder" object).  Consequently, shared memory namespace became filled with dependency results.
+    """
+    assert all_fully_qualified_parameter_names == [
+        "$parameter.raw.my_evaluation_dependency_parameter_name_1",
```

Comment:
@donaldheppner We could do this, but I do not think that we should.  My reasoning is as follows: in the implementation, we use those constants in order for the code to follow any possible changes in the syntax.  However, in tests, we want "to be literal" so that if something is wrong with the implementation, the tests will catch it.  Having said that, if you have a good argument for generalizing these strings, I would be happy to make the change.  Thanks.

Reply:
The value in referencing the constants is that the test becomes less brittle, and I can't think of a scenario where referencing constants here would fix an implementation error.

Reply:
@donaldheppner Change made (committed) -- please re-review.  Thank you!